File: Dummy/hawkshaw/library/ktextensions/CollectionsKt.java
package com.hawkshaw.library.ktextensions;

import X2.k;
import X2.o;
import Y1.K;
import java.util.Arrays;
import java.util.Collection;
import java.util.List;

public final class CollectionsKt {
    public static final <T> List<T> nullIfEmpty(List<? extends T> list) {
        Collection collection = list;
        if (collection == null || collection.isEmpty()) {
            return null;
        }
        return list;
    }

    public static final <T> T[] plus(T[] tArr, T... tArr2) {
        K.n(tArr, "<this>");
        K.n(tArr2, "elements");
        int length = tArr.length;
        int length2 = tArr2.length;
        T[] copyOf = Arrays.copyOf(tArr, length + length2);
        System.arraycopy(tArr2, 0, copyOf, length, length2);
        K.k(copyOf);
        return copyOf;
    }

    public static final <T> Collection<T> plusSafe(Collection<? extends T> collection, T... tArr) {
        K.n(collection, "<this>");
        K.n(tArr, "elements");
        return o.a1(k.L(tArr), collection);
    }
}

File: app/src/main/java/com/hawkshaw/library/ktextensions/CollectionsKt.kt
package com.hawkshaw.library.ktextensions

/**
 * Utility functions for collections
 */

/**
 * Returns null if the list is empty, otherwise returns the original list
 */
fun <T> List<T>?.nullIfEmpty(): List<T>? {
    return if (this.isNullOrEmpty()) null else this
}

/**
 * Adds an array of elements to the current array and returns a new array
 */
fun <T> Array<T>.plus(vararg elements: T): Array<T> {
    return this + elements
}

/**
 * Safely adds elements to a collection and returns a new collection
 */
fun <T> Collection<T>.plusSafe(vararg elements: T): Collection<T> {
    return this + elements
} 